addons/mail/tests/test_mail_composer.py

```diff
@@ -261,3 +264,66 @@ def test_mail_mass_mode_compose_with_mso(self):
             values[self.partner_employee.id]['body_html'],
             'We must preserve (mso) comments in email html'
         )
+
+    @users('employee')
+    def test_mail_attachment_to_download_link(self):
+        """ Test that when mail size exceed the max email size limit, attachment are turned into download links added
+        at the end of the email content. """
+        attachment_size_in_bytes = 1024 * 128
+        datas = base64.b64encode((''.join('.' for _ in range(0, attachment_size_in_bytes))).encode())
+        attachment1_name = 'attachment1'
+        attachment2_name = 'attachment2'
+        attachments = self.env['ir.attachment'].sudo().create([{
+            'name': attachment1_name,
+            'res_name': 'test',
+            'res_model': self.test_record._name,
+            'res_id': self.test_record.id,
+            'datas': datas,
+        }, {
+            'name': attachment2_name,
+            'res_name': 'test',
+            'res_model': self.test_record._name,
+            'res_id': self.test_record.id,
+            'datas': datas,
+        }])
+        match_download_links = [re.compile(fr'.*<a.*href.*/web/content/([^?]*)\?.*access_token.*>.*{attachment_name}.*</a>', re.DOTALL)
+                                for attachment_name in [attachment1_name, attachment2_name]]
+
+        def send_mail(max_email_size):
+            self.mail_server_global.max_email_size = max_email_size
```

Comment:
You could force the mail server in the composer (mail_server_id is normally supported), so that you don't have to update both test servers :)